.github/workflows/WhenPR.yml

```diff
@@ -0,0 +1,42 @@
+name: Run Test Android 
+
+on:
+  pull_request:
+    types: [opened,reopened,edited,synchronize]
+    branches: [ "main" ]
+  workflow_dispatch:
+jobs:
+  Test:
+
+    runs-on: ubuntu-latest
+
+    
+    steps:
+      - name: Fail if draft PR
+        if: ${{ github.event.pull_request.draft }}
+        run: |
+          echo "Pull request is in draft state."
+          exit 1
+
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '17'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Test with Gradle
+        run: ./gradlew koverXmlReport
+      - name: Check coverage
+        run: |
+          coverage=$(grep '<counter type="LINE"' $GITHUB_WORKSPACE/app/build/reports/kover/report.xml | sed 's/.*covered="\([0-9]*\)".*/\1/')
```

Comment:
I'm pretty sure there are some actions that allows you to do this in a better way